### DISCUSSION TITLE
command `grep` killed on starting zsh

### DISCUSSION THREAD

**Discussion by arximughal** [ohmyzsh/ohmyzsh#12093] [Help] [answered]
I have been stuck on this issue for a few days now. Whenever I do any of the below things:
- load a new terminal session
- run `source ~/.zshrc`
- execute any other command that internally uses `grep`

I get this error: 
```
[4]    33874 killed     command grep -q -Fx "$zcompdump_revision" "$ZSH_COMPDUMP" 2> /dev/null
[20]    34195 killed
[20]    34198 killed
[19]    34392 killed
[19]    34395 killed
```

I have tried completely removing `zsh` and `oh-my-zsh` and reinstalling both of these and also any other dependencies and config files, but nothing seems to work. 

I am on MacOS 13.5.2 (22G91). Zsh version is `zsh 5.9 (x86_64-apple-darwin22.6.0)`. 
There's not related logs in SystemLog or anywhere else I could find.

**Comment by carlosala** [comment, marked as answer]
You shouldn't `source ~/.zshrc`. In case you want to restart the shell, you can use `omz reload` (or `exec zsh` if you are not in an oh-my-zsh env). You never need to manually source `.zshrc`.

**Reply by arximughal** [reply]
Even that doesn't do the trick, to be honest. I had a lot of engagement this month so I gave up on it entirely for now. I will try. to fix it once I get some time off! 

Thanks for your answer.
